### ISSUE TITLE
Support for crossing VSTS service boundaries via the Location Service

### ISSUE THREAD

**Issue by jmyersmsft** [Microsoft/vsts-node-api#49]
I'd like to write API clients for Feed and NuGet APIs (and probably contribute them to this repository eventually). However, these APIs live in the Packaging service and we often only have the TFS service URL (e.g. in the build agent). We can do hacky things like assume that if the url ends in .visualstudio.com, the packaging service is at .pkgs.visualstudio.com, but VSTS already has a mechanism for determining the correct URL for an API area on another service: the Location Service.

In the C# client, the VssConnection class handles all that magic. When you ask it for a NuGetHttpClient, you get back one initialized with the Packaging service URL, no matter which service's URL you originally gave to VssConnection.

**Comment by willsmythe** [comment]
The VSTS Node client needs to start using “resource area” as a way to get the correct base URL for a particular client. As @jmyersmsft mentions, our .NET clients use this and are able to find the correct base URL (when the client is acquired from VssConnection). 

This avoids developers hand-forming URLs or  assuming that a particular API will always live on a particular service.

A new “resource area”  API was just introduced on VSTS for discovery the base URLs for different areas. It returns the full list of resource areas across the service (all service instance types) and includes the base URL that clients should use. The starting point is the "TFS" base URL (i.e. account URL): https://fabrikam.visualstudio.com.

The result would be ...

```
var collectionUrl = "https://fabrikam.visualstudio.com";   // always start with the “TFS” form of the base URL

var connection = new vsts.WebApi(collectionUrl, creds);

var releaseClient = connection.getReleaseApi();    // base URL is  https://fabrikam.vsrm.visualstudio.com
var buildClient = connection.getBuildApi();        // base URL is  https://fabrikam.visualstudio.com
```

For clients not bound to a particular service instance type, (like Notifications), there should also be a way to specify the service instance type:

```
var notificationClient = connection.getNotificationApi(ServiceInstanceTypes.EMS);      // base URL is  https://fabrikam.extmgmt.visualstudio.com
```

**Comment by stephenmichaelf** [comment]
This has been resolved in #139.
